src/crimson/osd/pg.cc

```diff
@@ -936,8 +951,32 @@ PG::submit_transaction(
     projected_log.add(entry);
   }
 
+  std::vector<pg_shard_t> shards_to_send_txn;
```

Comment:
What do you think about:

`using pg_shard_should_send = std::pair<pg_shard_t, bool>`

`std::vector<pg_shard_should_send> shards_to_send_logs;`

This should make traversing easier later on

Reply:
Yeah, will do.